### PR TITLE
Some code cleanups

### PR DESCRIPTION
**Pull request by gkorland**
1. diamonds operator
2. missing final
3. index of with '' when possible

### CODE REVIEW

**Comment by gkorland** [comment]
@sazzad16 but there is no need every nested enum is static

**Comment by gkorland** [comment]
@sazzad16 any more comments but the `static enum`?

File: src/main/java/redis/clients/jedis/util/Pool.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
   public Pool() {
   }
 
-  public Pool(final GenericObjectPoolConfig poolConfig, PooledObjectFactory<T> factory) {
+  public Pool(final GenericObjectPoolConfig<T> poolConfig, PooledObjectFactory<T> factory) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/util/Pool.java]: Won't this create backward compatibility issue?

File: src/main/java/redis/clients/jedis/util/Pool.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public boolean isClosed() {
     return this.internalPool.isClosed();
   }
 
-  public void initPool(final GenericObjectPoolConfig poolConfig, PooledObjectFactory<T> factory) {
+  public void initPool(final GenericObjectPoolConfig<T> poolConfig, PooledObjectFactory<T> factory) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/util/Pool.java]: and this?

**Comment by sazzad16** [comment]
@gkorland Except `static enum`? No, there aren't anything that I object.
Although I'm a bit concerned about `GenericObjectPoolConfig` -> `GenericObjectPoolConfig<T>` change. We may have to separate those changes for 4.0+.

**Comment by gkorland** [comment]
I guess we can close this PR since we just merged #2100

**Comment by sazzad16** [comment]
@gkorland We could go further with one. #2100 has 64 changes. But this PR had many more.

**Comment by gkorland** [comment]
I'll submit  a new PR

**Comment by sazzad16** [comment]
okay

**Pull request closed without merging**